src/rust-crypto/rust-crypto.ts

```diff
@@ -615,7 +615,31 @@ export class RustCrypto extends TypedEventEmitter<RustCryptoEvents, RustCryptoEv
      * Implementation of {@link CryptoApi#isSecretStorageReady}
      */
     public async isSecretStorageReady(): Promise<boolean> {
-        return false;
+        const secretStorageKeyTuple = await this.secretStorage.getKey();
+        if (!secretStorageKeyTuple) return false;
+
+        const [keyId] = secretStorageKeyTuple;
+
+        // Get the master signing keys stored into the secret storage
```

Comment:
This doesn't actually get the master key itself. Rather, it gets the key that is used to encrypt the master key.

```suggestion
        // Get the keys that were used to encrypt the master cross-signing key in secret storage
```

likewise the others.